src/Resources/config/menu.xml

```diff
@@ -20,10 +20,6 @@
         <service id="sonata.admin.menu.matcher.voter.admin" class="Sonata\AdminBundle\Menu\Matcher\Voter\AdminVoter" public="true">
             <tag name="knp_menu.voter" request="true"/>
         </service>
-        <service id="sonata.admin.menu.matcher.voter.children" class="Sonata\AdminBundle\Menu\Matcher\Voter\ChildrenVoter" public="true">
```

Comment:
This is a public service, so a BC break

Reply:
Technically yes

Reply:
getting it from the container before getting the menu matcher was triggering an exception due to a circular reference previously AFAICT.

Anyway, for full BC, you could keep the service and mark it as deprecated, if you just remove the tag (so that it does not get registered itself in the matcher, which is causing the bugs being resolved by the PR)

Reply:
In that case it sounds like the workflow xkcd though

![image](https://user-images.githubusercontent.com/657779/33381076-3e4f3f8e-d51d-11e7-82b7-e9d13d601783.png)

Reply:
`@deprecated` is supported now since we bumped to 2.8 though, so I'm going to do that.

Reply:
> In that case it sounds like the workflow xkcd though

I knew that you would post this xkcd :trollface:

Reply:
Sorry couldn't resist :P